lib/OpenQA/Shared/Controller/Session.pm

```diff
@@ -25,8 +25,14 @@ sub ensure_operator {
 
 sub ensure_admin {
     my ($self) = @_;
-    $self->redirect_to($self->url_for('login')->query(return_page => $self->req->url)) and return undef
-      unless $self->current_user;
+    unless ($self->current_user) {
+        if ($self->tx->req->headers->header('Accept') eq 'application/json') {
```

Comment:
`Use of uninitialized value in string eq at /home/squamata/project/lib/OpenQA/Shared/Controller/Session.pm line 29` from t/03-auth.t

Reply:
```suggestion
        if (($self->tx->req->headers->accept // '') eq 'application/json') {
```